src/components/common/CodeReviewToolbar.tsx

```diff
@@ -0,0 +1,274 @@
+  TOOLBAR_EDIT_CODE_BUTTON_CYPRESS,
+  TOOLBAR_RUN_CODE_BUTTON_CYPRESS,
+  TOOLBAR_USER_SELECT_CYPRESS,
+  TOOLBAR_VERSION_SELECT_CYPRESS,
+  TOOLBAR_VISIBILITY_BUTTON_CYPRESS,
+} from '../../config/selectors';
+import { DEFAULT_LINE_HIDDEN_STATE } from '../../config/settings';
+import { CodeVersionSelectType } from '../../interfaces/codeVersions';
+import { SETTINGS_KEYS } from '../../interfaces/settings';
+import { NO_DATE_PLACEHOLDER, getFormattedTime } from '../../utils/datetime';
+import { useCodeVersionContext } from '../context/CodeVersionContext';
+import { useSettings } from '../context/SettingsContext';
+import { useVisibilityContext } from '../context/VisibilityContext';
+import CustomDialog from '../layout/CustomDialog';
+import CustomSelect from '../layout/CustomSelect';
+import ToolbarButton from '../layout/ToolbarButton';
+import CommitInfo from './CommitInfo';
+
+// generate the labels
+const getVersionLabel = (
+  { data, updatedAt }: CodeVersionSelectType,
+  i18n: i18nType,
+): string => {
+  const { commitMessage } = data;
+  let msg = commitMessage || i18n.t('noCommitMessage');
+  // if message id too long: truncate and add ellipsis
+  if (msg.length > DEFAULT_TRUNCATION_COMMIT_MESSAGE_LENGTH) {
+    msg = `${commitMessage.slice(
+      0,
+      DEFAULT_TRUNCATION_COMMIT_MESSAGE_LENGTH,
+    )}…`;
+  }
+  // format createdAt date
+  // a placeholder is used if the property does not exist (fake API)
+  const date = updatedAt
+    ? getFormattedTime(updatedAt, i18n.language)
+    : NO_DATE_PLACEHOLDER;
+  return `${msg} - ${date}`;
+};
+
+const StyledStack = styled(Stack)(({ theme }) => ({
+  marginBottom: theme.spacing(2),
+}));
+
+type Props = {
+  setView: (view: AppView) => void;
+};
+
+const CodeReviewToolbar: FC<Props> = ({ setView }) => {
+  const { t, i18n } = useTranslation();
+  const { settings } = useSettings();
+  const { toggleAll } = useVisibilityContext();
+  const {
+    groupedVersions,
+    setCodeId,
+    codeVersion,
+    codeId,
+    codeVersionResource,
+  } = useCodeVersionContext();
+  const showToolbar = settings[SETTINGS_KEYS.SHOW_TOOLBAR];
+  const showEditButton = settings[SETTINGS_KEYS.SHOW_EDIT_BUTTON];
+  const showVisibilityToggle = settings[SETTINGS_KEYS.SHOW_VISIBILITY_BUTTON];
+  const showCommitInfo = settings[SETTINGS_KEYS.SHOW_VERSION_NAVIGATION];
+  const isExecutable = SUPPORTED_EXECUTABLE_LANGUAGES.includes(
+    codeVersion.language,
+  );
+  const [isOpenCommitInfo, setIsOpenCommitInfo] = useState(false);
+  const [isHidden, setIsHidden] = useState(DEFAULT_LINE_HIDDEN_STATE);
+  const [selectedUser, setSelectedUser] = useState(
+    groupedVersions[0].user.value,
+  );
+
+  const getFormattedVersionOptions = (
+    versions: CodeVersionSelectType[],
+  ): { label: string; value: string }[] =>
+    versions.map((v) => ({
+      label: getVersionLabel(v, i18n),
+      value: v.id,
+    })) || [{ label: INSTRUCTOR_CODE_NAME, value: INSTRUCTOR_CODE_ID }];
```

Comment:
Looks like you could take it out of the component?

Reply:
I am still using `i18n` value for the hook inside the function, so maybe i keep it inside, as i do not really like to have to pass i18n every time i call the function.